src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinHttpWebSocket.cs

```diff
@@ -441,6 +448,12 @@ private Task<bool> InternalReceiveAsync(ArraySegment<byte> buffer)
                 Abort();
             }
 
+            // If the task has completed, there's no need to force continuations to run async
+            if (_operation.TcsReceive.Task.Status == TaskStatus.RanToCompletion)
+            {
+                return Task.FromResult(_operation.TcsReceive.Task.GetAwaiter().GetResult());
+            }
```

Comment:
What is the consuming code doing that this pattern helps?  Is it using ContinueWith(..., TaskContinuationOptions.ExecuteSynchronously) off of the returned task, and you want that continuation to be invoked as part of the ContinueWith call rather than asynchronously from the ContinueWith call?